extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -215,6 +226,21 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
       this.builtTemplates_[templateId] = template;
       this.mutateElement(() => this.templateContainer_.appendChild(template));
     }
+
+    // PDP view that calls the analytics service sending the product id.
+    if (productForPdp) {
+      this.variableService_.onVariableUpdate(
+        AnalyticsVariable.STORY_SHOPPING_PRODUCT_ID,
+        productForPdp.productId
+      );
+    }
+
+    // Triggers an analytics event for the PDP or PLP depending on if there is an active product.
+    this.analyticsService_.triggerEvent(
+      StoryAnalyticsEvent[
+        productForPdp ? 'SHOPPING_PDP_VIEW' : 'SHOPPING_PLP_VIEW'
```

Comment:
We want the compiler to be able to optimize this code, and not need to do unnecessary dictionary lookups. `productForPdp ? StoryAnalyticsEvent.SHOPPING_PDP_VIEW : StoryAnalyticsEvent.SHOPPING_PLP_VIEW`